test/examples/api_fam_add.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * api_fam_add.cpp
+ * Copyright (c) 2020 Hewlett Packard Enterprise Development, LP. All rights
+ * reserved. Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+
+#include <fam/fam.h>
+#include <fam/fam_exception.h>
+#include <string.h>
+using namespace std;
+using namespace openfam;
+
+int main(void) {
+    int ret = 0;
+    fam *myFam = new fam();
+    Fam_Options *fm = (Fam_Options *)malloc(sizeof(Fam_Options));
+    memset((void *)fm, 0, sizeof(Fam_Options));
+    // assume that no specific options are needed by the implementation
+    fm->runtime = strdup("NONE");
+    try {
+        if (myFam->fam_initialize("myApplication", fm) == 0) {
+            printf("FAM initialized\n");
+        }
+    } catch (Fam_Exception &e) {
+        printf("FAM Initialization failed: %s\n", e.fam_error_msg());
+        myFam->fam_abort(-1); // abort the program
+        // note that fam_abort currently returns after signaling
+        // so we must terminate with the same value
+        return -1;
+    }
+
+    // ... Initialization code here
```

Comment:
Can examples run as standalone programs ? Do you want to add fam_create_region and fam_allocate to all the examples ? Or do you want to document it in readme.md?